backend/src/main/java/com/example/demo/service/DemoService.java

```diff
@@ -71,8 +71,16 @@ public CommonResponse patchUserWidget(String email, UserWidgetDto[] userWidgetDt
 
         for (UserWidgetDto userWidgetDto : userWidgetDtos) {
             Widget findExistWidget = widgetRepository.findByName(userWidgetDto.getName());
-            if (findExistWidget == null)
-                System.out.println(responseService.errorResponse(400, "widget not found").log);
+
+            if (findExistWidget == null) {
+                Widget widget = new Widget();
+                if (userWidgetDto.getName() == null)
+                    return responseService.errorResponse(400, "widget name null");
```

Comment:
```suggestion
                if (userWidgetDto.getName() == null)
                    return responseService.errorResponse(400, "widget name null");
                Widget widget = new Widget();
```
`userWidgetDto`의 이름이 null이더라도 Widget이 그대로 생성되니 확인 구문을 위로 올리면 좋을 것 같습니다.

Reply:
그렇게 변경하면 좀 더 효율적일 수 있겠네요! 감사합니다.